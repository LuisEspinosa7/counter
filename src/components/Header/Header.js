import React from "react";
import { StyledFlexContainer } from "../styles/containers/FlexContainer.styled";
import { StyledHeaderContainer } from "../styles/containers/HeaderContainer.styled";
import {
  StyledIcon,
  StyledIconTitle,
  StyledIconWrapper,
} from "../styles/icons/Icons.styled";

const Header = () => {
  return (
    <StyledHeaderContainer gridArea="navbar" justifyContent="space-between">
      <StyledFlexContainer
        justifyContent="flex-start"
        verticalPadding="0.4rem"
        horizontalPadding="1.5rem"
      >
        <StyledIconWrapper
            width="4.25rem"
            height="4.25rem">
          <StyledIcon src="./images/icon_number_white.svg" />
        </StyledIconWrapper>
        <StyledIconTitle
          fontSize="2rem"
          fontWeight="800"
          verticalPadding="0"
          horizontalPadding="0.8rem"
        >
          Counter
        </StyledIconTitle>
      </StyledFlexContainer>

      <StyledIconWrapper
        verticalMargin="0"
        horizontalMargin="0.8rem"
        width="50px"
        height="50px"
      >
        <StyledIcon
          src="./images/hamburger_icon_white.svg"
          rotateDegree="-180deg"
        />
      </StyledIconWrapper>
    </StyledHeaderContainer>
  );
};

export default Header;
