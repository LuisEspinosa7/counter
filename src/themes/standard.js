export const theme = {
    colors: {
        primary: {
            normal: '#4764e6',
            light: '#398AB9',
            dark: '#145071'
        },
        secondary: {
            normal: '#f2f4ff'
        },
        accent: {
            normal: '#4bd6f2'
        },
        gray: {
            normal: '#EEEEEE',
            dark: '#e6e4e4'
        },
        white: {
            normal: '#fff'
        },
        black: {
            normal: '#000'
        }
    },
    mainContainer: {
        maxWidth: '1600px'
    }
}