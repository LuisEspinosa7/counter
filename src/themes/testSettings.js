import { ThemeConsumer } from 'styled-components';
import { shallow } from 'enzyme/build';
import { theme } from './standard';


/* This is an approach to solve the enzyme tests when using a
   Styled components Theme
 */
export const shallowWithTheme = (children, them = theme) => {
    ThemeConsumer._currentValue = them;
    return shallow(children);
}