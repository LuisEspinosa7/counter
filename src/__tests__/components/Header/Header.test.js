import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-enzyme';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../../themes/standard';
import { render } from 'enzyme/build';
import { shallowWithTheme } from '../../../themes/testSettings';
import Header from '../../../components/Header/Header';


describe("Component: Header", () => {

    it('should render', () => {
        /* We use render here because we only need to take the snapshot */
        const { fragment } = render(
            <ThemeProvider theme={theme}>
                <Header />
            </ThemeProvider>
        )
        const tree = renderer.create(fragment).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('should have 2 <i> elements and 1 <h2>', () => {
        const wrapper = shallowWithTheme(<Header />);
        //console.log(wrapper.debug());
        expect(wrapper.find('i')).toHaveLength(2);
        expect(wrapper.find('h2')).toHaveLength(1);
        expect(wrapper.find('h2').text()).toContain('Counter');
    })

})