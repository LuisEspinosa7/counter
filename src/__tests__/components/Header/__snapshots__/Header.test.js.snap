// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Header should render 1`] = `null`;
